Libraries/xcsdk/Sources/SDK/Target.cpp

```diff
@@ -61,19 +63,50 @@ executablePaths() const
 bool Target::
 parse(plist::Dictionary const *dict)
 {
-    auto V    = dict->value <plist::String> ("Version");
-    auto CN   = dict->value <plist::String> ("CanonicalName");
-    auto DN   = dict->value <plist::String> ("DisplayName");
-    auto MDN  = dict->value <plist::String> ("MinimalDisplayName");
-    auto MDT  = dict->value <plist::String> ("MaximumDeploymentTarget");
-    auto SBTV = dict->value <plist::Array> ("SupportedBuildToolsVersion");
-    auto CP   = dict->value <plist::Dictionary> ("CustomProperties");
-    auto DP   = dict->value <plist::Dictionary> ("DefaultProperties");
-    auto IBS  = dict->value <plist::String> ("IsBaseSDK");
-    auto TCV  = dict->value <plist::Array> ("Toolchains");
+    std::unordered_set<std::string> seen;
+    auto unpack = plist::Keys::Unpack("Target", dict, &seen);
+
+    auto Ts    = unpack.cast <plist::Array> ("Toolchains");
+    auto CN    = unpack.cast <plist::String> ("CanonicalName");
+    auto DN    = unpack.cast <plist::String> ("DisplayName");
+    auto MDN   = unpack.cast <plist::String> ("MinimalDisplayName");
+    auto V     = unpack.cast <plist::String> ("Version");
+    auto IBS   = unpack.coerce <plist::Boolean> ("IsBaseSDK");
+    auto DDT   = unpack.cast <plist::String> ("DefaultDeploymentTarget");
+    auto MDT   = unpack.cast <plist::String> ("MaximumDeploymentTarget");
+    auto DP    = unpack.cast <plist::Dictionary> ("DefaultProperties");
+    auto CP    = unpack.cast <plist::Dictionary> ("CustomProperties");
+    auto PCFNs = unpack.cast <plist::Array> ("PropertyConditionFallbackNames");
+    auto DSFN  = unpack.cast <plist::String> ("DocSetFeedName");
+    auto DSFU  = unpack.cast <plist::String> ("DocSetFeedURL");
+    auto MSTV  = unpack.cast <plist::String> ("MinimumSupportedToolsVersion");
+    auto SBTCs = unpack.cast <plist::Array> ("SupportedBuildToolComponents");
+
+    /* Ignored: seems to be a typo. */
+    (void)unpack.cast <plist::String> ("isBaseSDK");
```

Comment:
Why all of these ignored unpackings?  I assume unpack.complete() won't return true otherwise?